### PR TITLE
properties for Z

### PR DESCRIPTION
**Pull request by 3pmile**
**Description**

This PR implements...
- [x] is_one for Z, PolyOverZ
- [x] is_identity MatZ, MatPolyOverZ
- [x] is_zero for Z, MatZ, PolyOverZ, MatPolyOverZ
- [x] is_squared  for MatZ, MatPolyOverZ

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/integer/z/properties.rs [hunk 1]
```diff
@@ -219,3 +247,51 @@ mod test_inv {
...
+        let zero = Z::from(0);
+
+        assert!(zero.is_zero());
+    }
+
+    /// ensure that is_zero returns `false` for non-zero values
+    #[test]
+    fn zero_rejection() {
+        let small = Z::from(2);
+        let large = Z::from_str(&format!("{}", (u128::MAX - 1) / 2 + 1)).unwrap();
```
Comment (jnsiemer) [inline src/integer/z/properties.rs:268]:
```suggestion
        let large = Z::from(u64::MAX);
```
suffices for large integer testing as its bigger than 2^62. Please remove this unnecessarily complex part also at all other parts in this PR.

**Review by jnsiemer — CHANGES_REQUESTED** [review]
Please apply the suggestions also to the other open PRs for Zq and Q

File: src/integer/mat_poly_over_z/properties.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 use super::MatPolyOverZ;
 use crate::traits::{GetNumColumns, GetNumRows};
-use flint_sys::fmpz_poly_mat::fmpz_poly_mat_is_one;
+use flint_sys::fmpz_poly_mat::{fmpz_poly_mat_is_one, fmpz_poly_mat_is_zero};
 
 impl MatPolyOverZ {
```
[the remark this answers is not in the document]
Comment (3pmile) [reply src/integer/mat_poly_over_z/properties.rs:16]: its already implemented

File: src/integer/z/properties.rs (hunk 1 above)
Comment (3pmile) [reply src/integer/z/properties.rs:268]: this is wanted, it is the least complex representation of 2^n with n>62 so that the last bits are all zero

**Pull request closed without merging**